Modules/.AL-Go/CompileAppInBcContainer.ps1

```diff
@@ -2,40 +2,43 @@ Param(
     [Hashtable] $parameters
 )
 
+# Setup compiler features to generate captions and LCGs
 if (!$parameters.ContainsKey("Features")) {
     $parameters["Features"] = @()
 }
-$parameters["Features"] = @("lcgtranslationfile", "generateCaptions")
+$parameters["Features"] += @("lcgtranslationfile", "generateCaptions")
 
 $appFile = Compile-AppInBcContainer @parameters
 
-if ($appFile) {
-    $filename = [System.IO.Path]::GetFileName($appFile)
-    if ($filename -like "Microsoft_System Application_*.*.*.*.app") {
-        # System application compiled - add BaseApp and Application app from container to output
-        Invoke-ScriptInBcContainer -containerName $parameters.ContainerName -scriptblock { Param([string]$packagesFolder)
-            function CopyApp {
-                Param(
-                    [string] $localAppPath,
-                    [string] $w1AppPath
-                )
-
-                $name = [System.IO.Path]::GetFileName($w1AppPath)
-                Write-Host "Copying $name to packages path"
-                if (Test-Path $localAppPath) {
-                    $appPath = $localAppPath
-                }
-                else {
-                    $appPath = $w1AppPath
-                }
-                Copy-Item -Path $appPath -Destination (Join-Path $packagesFolder $name)
-                Copy-Item -Path $appPath -Destination (Join-Path "c:\run\my" $name)
-            }
-
-            CopyApp -localAppPath "C:\Applications.*\Microsoft_Base Application_*.*.*.*.app"    -w1AppPath "C:\Applications\BaseApp\Source\Microsoft_Base Application.app"
-            CopyApp -localAppPath "C:\Applications.*\Microsoft_Application_*.*.*.*.app"         -w1AppPath "C:\Applications\Application\Source\Microsoft_Application.app" 
-        } -argumentList (Get-BcContainerPath -ContainerName $parameters.ContainerName -path $Parameters.appSymbolsFolder)
-    }
+$appProjectFolder = $parameters.appProjectFolder
+
+# Extract app name from app.json
+$appName = (gci -Path $appProjectFolder -Filter "app.json" | Get-Content | ConvertFrom-Json).name
+
+Write-Host "Current app name: $appName; app folder: $appProjectFolder"
+
+# Create an archive with the current source code in the build artifacts folder
+
+$archiveFile = "$env:TEMP/$appName.Source.zip"
+Write-Host "Archive the current source code for app: $appName as $archiveFile"
+Compress-Archive -Path "$appProjectFolder/**" -DestinationPath $archiveFile -Force
```

Comment:
Have you considered storing the source unzipped?
- The nupkg format is already zipped, so zip files inside do not save any space.
- It would be simpler to consume.

Reply:
This step is to transfer the source code to GitHub artifacts so that it can be used in the _Deliver_ task. So, it's more about saving space there and not have to deal with files and folders but with a simple .zip.

We'll re-think which approach suits us best.